### PR TITLE
Fix https://github.com/lesfurets/git-octopus/issues/34

### PR DESCRIPTION
**Pull request by danoliv**
This change is [Reviewable](https://reviewable.io/reviews/lesfurets/git-octopus/35)

### CODE REVIEW

File: run/basics_test.go [hunk 1]
```diff
@@ -84,3 +84,29 @@ func TestOctopus3branches(t *testing.T) {
...
+	repo.Git("rev-parse", "HEAD")
+	repo.Git("checkout", "master")
+
+	//when
+	Run(context, "feature/*")
+
+	//then
+	// feature/test should be merged in master
+	actual, _ := repo.Git("branch", "--contains", "feature/test")
+	assert.Contains(t, actual, "master")
```
Comment (apflieger) [inline run/basics_test.go:107]: Given that the fast forward logic has been removed, this test should assert that the behavior is still working. Yeah.. it wasn't tested originally.
You could replace this assertion with something more precise:
```go
repo.Git("commit", "-m", "add testFeature")
testFeatureSha1, _ := repo.Git("rev-parse", "HEAD")

...
// Assert that master has been fast forwarded to feature/test
masterSha1, _ := repo.Git("rev-parse", "HEAD")
assert.Equals(t, testFeatureSha1, masterSha1)
```

**Comment by danoliv** [comment]
I fixed the test as you said, I restored the fastforward logic. But read-tree was doing nothing in the fastforward case, even if it should have accordingly to documentation. 

I end up using a merge command instead of read-tree in the fastforward case.

**Comment by apflieger** [comment]
I gave you push rights to the repo so we can iterate on the same branch.

A bit of context : the `mergeHeads` function is a copy of the octopus merge strategy from git [here](https://github.com/git/git/blob/master/git-merge-octopus.sh). This function, as the script, is not supposed to do any ref updates (which git merge does). The change might work, I thought about a conter example but didn't find any. I'd still prefer to keep having the same logic that git-merge-octopus.sh has. The fix should be on the caller of the function.

I just pushed a pull request with a fix so we can talk about it #36

**Comment by apflieger** [comment]
Fixed by #36

**Pull request merged**